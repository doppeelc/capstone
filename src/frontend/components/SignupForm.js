import React, { useState, useEffect } from "react";
import "../styles/SignupForm.css";


function SignupForm({ signup }) {

    const INIT_STATE = {
        username:"",
        password:"",
        displayName:"",
        email:""
    }
    
    const [formData, setFormData] = useState(INIT_STATE);

    function handleChange(e) {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]:value
        }));
    }

    function handleSubmit(e) {
        e.preventDefault();
        signup(formData);
        setFormData(INIT_STATE);
    }

    return (
        <div className="SignupForm" >
            <h1>Sign Up</h1>
            <form id="SignupForm" onSubmit={handleSubmit}>
                <label id="lbl" htmlFor="username">Username</label>
                <input
                    id="inpt"
                    name="username"
                    placeholder="username"
                    value={formData.username}
                    onChange={handleChange} />
                
                <label id="lbl" htmlFor="password">Password</label>
                <input
                    id="inpt"
                    name="password"
                    placeholder="password"
                    value={formData.password}
                    onChange={handleChange} />
                    
                <label id="lbl" htmlFor="displayName">Display Name</label>
                <input
                    id="inpt"
                    name="displayName"
                    placeholder="display name"
                    value={formData.displayName}
                    onChange={handleChange} />
                
                <label id="lbl" htmlFor="email">Email</label>
                <input
                    id="inpt"
                    name="email"
                    placeholder="email"
                    value={formData.email}
                    onChange={handleChange} />
                <button>Sign Up</button>
            </form>
        </div>
    );
}

export default SignupForm;