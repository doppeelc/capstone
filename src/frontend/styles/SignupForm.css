.SignupForm {
    width: 13%;
    margin: auto;
}