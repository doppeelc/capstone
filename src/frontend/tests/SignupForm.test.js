import { render } from '@testing-library/react';
import SignupForm from '../components/SignupForm';
import UserContext from '../components/UserContext';
import { currentUser } from './__testCommon';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';

test('renders without crashing', () => {
    render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser:currentUser}}>
                <SignupForm signup={() => {}}/>
            </UserContext.Provider>
        </BrowserRouter>
    );
});

test('matches the snapshot', () => {
    const { asFragment } = render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser:currentUser}}>
                <SignupForm signup={() => {}}/>
            </UserContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('renders the form empty', () => {
    const { getByPlaceholderText } = render(<SignupForm />);
    expect(getByPlaceholderText("username")).toBeInTheDocument();
    expect(getByPlaceholderText("username")).toBeEmptyDOMElement();
    expect(getByPlaceholderText("password")).toBeInTheDocument();
    expect(getByPlaceholderText("password")).toBeEmptyDOMElement();
    expect(getByPlaceholderText("email")).toBeInTheDocument();
    expect(getByPlaceholderText("email")).toBeEmptyDOMElement();
    expect(getByPlaceholderText("display name")).toBeInTheDocument();
    expect(getByPlaceholderText("display name")).toBeEmptyDOMElement();
});






