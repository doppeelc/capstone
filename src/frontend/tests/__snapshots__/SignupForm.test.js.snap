// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <h1>
      Sign Up
    </h1>
    <form
      id="SignupForm"
    >
      <label
        for="username"
        id="lbl"
      >
        Username
      </label>
      <input
        id="inpt"
        name="username"
        placeholder="username"
        value=""
      />
      <label
        for="password"
        id="lbl"
      >
        Password
      </label>
      <input
        id="inpt"
        name="password"
        placeholder="password"
        value=""
      />
      <label
        for="displayName"
        id="lbl"
      >
        Display Name
      </label>
      <input
        id="inpt"
        name="displayName"
        placeholder="display name"
        value=""
      />
      <label
        for="email"
        id="lbl"
      >
        Email
      </label>
      <input
        id="inpt"
        name="email"
        placeholder="email"
        value=""
      />
      <button>
        Sign Up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
